packages/vaex-plotly/vaex/plotly/__init__.py

```diff
@@ -0,0 +1,593 @@
+                customdata = None
+            hovertemplate += '<extra></extra>'
+
+            # the plotting starts here
+            marker = go.scatter.Marker(color=color_values, size=size_values, showscale=colorbar,
+                                       colorscale=colormap, symbol=symbol_value, colorbar=cbar)
+
+            trace = go.Scatter(x=x_values, y=y_values, error_x=xerr_object, error_y=yerr_object,
+                               mode='markers',
+                               marker=marker,
+                               hovertemplate=hovertemplate,
+                               customdata=customdata,
+                               hovertext=hover_title,
+                               name=label_value)
+            traces.append(trace)
+
+        legend = go.layout.Legend(orientation='h')
+        title = go.layout.Title(text=title, xanchor='center', x=0.5, yanchor='top')
+        layout = go.Layout(height=figure_height,
+                           width=figure_width,
+                           legend=legend,
+                           title=title,
+                           xaxis=go.layout.XAxis(title=xlabel or x[0]),
+                           yaxis=go.layout.YAxis(title=ylabel or y[0],
+                                                 scaleanchor='x',
+                                                 scaleratio=1))
+
+        fig = go.FigureWidget(data=traces, layout=layout)
+
+        # Define the widget components
+        _widget_selection = widgets.ToggleButtons(options=['default'], description='selection')
+        _items = [{'text': xexpr + ' -vs-  ' + yexpr, 'value': i} for i, (xexpr, yexpr) in enumerate(zip(x, y))]
+        _widget_selection_space = vue.Select(items=_items, v_model=0, label='selection space')
+        _widget_selection_mode = widgets.ToggleButtons(options=['replace', 'and', 'or', 'xor', 'subtract'],
+                                                       value='replace',
+                                                       description='mode')
+        _widget_selection_undo = widgets.Button(description='undo', icon='arrow-left')
+        _widget_selection_redo = widgets.Button(description='redo', icon='arrow-right')
+        _widget_history_box = widgets.HBox(children=[widgets.Label('history', layout={'width': '80px'}),
+                                                     _widget_selection_undo,
+                                                     _widget_selection_redo])
+        # Put them together in the control-widget: this is what is contained within the navigation drawer
+        control_widget = vue.Layout(pa_1=True, column=True, children=[_widget_selection_space,
+                                                                      _widget_selection,
+                                                                      _widget_selection_mode,
+                                                                      _widget_history_box])
+        # The output widget
```

Comment:
this comments is superfluous, it does not convey information. A better comment would be the reason why you have this widget.
```suggestion
        # An output widget is used to capture output during callbacks
```